chapters/data-analysis.tex

```diff
@@ -1,292 +1,488 @@
+It is often useful to start looking at comparisons and other documentation 
+outside the code editor.
+
+Adding comments to the code explaining what you are doing and why is crucial here.
+There are always ways for things to go wrong that you never anticipated, but two issues to pay extra attention to are missing values and dropped observations. 
+If you are subsetting a data set, you should drop observations explicitly, indicating why you are doing that and how the data set changed.
+Merging, reshaping and aggregating data sets can change both the total number of observations and the number of observations with missing values.
+Make sure to read about how each command treats missing observations and, whenever possible, add automated checks in the script that throw an error message if the result is changing.
+
+At this point, you will also need to address some of the issues in the data that you identified during data cleaning. 
+The most common of them is the presence of outliers.
+How to treat outliers is a research question, but make sure to note what we the decision made by the research team, and how you came to it. 
+Results can be sensitive to the treatment of outliers, so keeping the original variable in the data set will allow you to test how much it affects the estimates.
+More generally, create derived measures in new variables instead of overwriting the original information.
+
+% Outputs -----------------------------------------------------------------
+
+% Data set
+The outputs of construction are the data sets that will be used for analysis.
+The level of observation of a constructed data set is the unit analysis. 
+Each data set is purpose-built to answer an analysis question.
+Since different pieces of analysis may require different samples,
+or even different units of observation,
+you may have one or multiple constructed data sets, 
+depending on how your analysis is structured.
+So don't worry if you cannot create a single, ``canonical'' analysis data set.
+It is common to have many purpose-built analysis datasets:
 there may be a \texttt{data-wide.dta},
 \texttt{data-wide-children-only.dta}, \texttt{data-long.dta},
 \texttt{data-long-counterfactual.dta}, and many more as needed.
-Data construction should never be done in analysis files,
-and since much data construction is specific to the planned analysis,
-organizing analysis code into many small files allows that level of specificity
-to be introduced at the correct part of the code process.
-Then, when it comes time to finalize analysis files,
-it is substantially easier to organize and prune that code
-since there is no risk of losing construction code it depends on.
+One thing all constructed data sets should have in common, though,
+are functionally-named variables.
+As you no longer need to worry about keeping variable names
+consistent with the survey, they should be as intuitive as possible.
```

Comment:
_As you no longer need to worry about keeping variable names consistent with the survey, they should be as intuitive as possible._ - We write _no longer worry_ as if it doesn't matter. It is great if they are when it can be combined, but it is more important that it is intuitive then matching, so it is not necessary that they are the same anymore, but we should not write this as if it doesn't matter at all.

Reply:
>One thing all constructed data sets should have in common, though, are functionally-named variables.
>Constructed variables are called ``constructed'' because they were not present in the survey to start with,
so making their names consistent with the survey form is not as crucial.
>Of course, whenever possible, having variables names that are both intuitive and can be linked to the survey is ideal.
>However, functionality should be prioritized here.

(c80a095)